firebase-firestore/src/main/java/com/google/firebase/firestore/local/LocalDocumentsView.java

```diff
@@ -316,7 +317,31 @@ void recalculateAndSaveOverlays(Set<DocumentKey> documentKeys) {
     }
   }
 
-  /** Queries the remote documents and overlays by doing a full collection scan. */
+  /**
+   * Returns a mapping of documents that match the provided query and offset mapped to the
+   * corresponding batch id.
+   */
+  // TODO(indexing): Support adding local mutations to collection group table.
+  public Map<Document, Integer> getDocumentsBatchIdsMatchingCollectionGroupQuery(
```

Comment:
Your code would be cleaner if this method returned `Map<BatchId, List<Document>>`.

The name is also quite a mouthful - and the method should probably just accept a `String collectionGroup` rather than a query as it ignores all other query constraints.